### ISSUE TITLE
Allow long URLs migration fails upgrading wagtail to 2.13.3

### ISSUE THREAD

**Issue by marts** [wagtail/wagtail#7323]
### Issue Summary

Hello, I'm upgrading an existing Wagtail site 2.7.3 to 2.13.3. All worked fine on my test database, but when running the migrations on production data I encountered the following error when `wagtailembeds.0008_allow_long_urls` was applied:

```
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(200)
```

After a bit of digging, I found that my production data has a couple of Embed objects with `thumbnail_url` lengths greater than 200.

This seemed OK, as Wagtail 2.7.3 has `thumbnail_url` max length of 255:

```
thumbnail_url = models.URLField(max_length=255, null=True, blank=True)
```

I tried a quick fix by updating the `wagtailembeds.0008_allow_long_urls` migration by adding `max_length=255`. It seemed to fix the problem and the migration ran OK on my production data.

However, I'm not sure if this is the preferred or right solution:

```
migrations.AlterField(
    model_name="embed",
    name="thumbnail_url",
    field=models.URLField(
        blank=True,
        default="",
        max_length=255,
    ),
    preserve_default=False,
),
```

### Steps to Reproduce

I have only tried this on a existing Wagtail instance, but I suppose it could be recreated on a fresh (but older 2.7.3?) instance of Wagtail, creating an Embed object with a `thumbnail_url` greater that 200 in length and then migrating to the latest Wagtail (2.13.3).

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: No

### Technical details

* Python version: Python 3.6.4
* Django version: 3.2.5
* Wagtail version: 2.13.3
* Browser version: N/A

**Comment by gasman** [comment]
Thanks for the report @marts! I think your suggested fix is correct - the AlterField operation in question is only needed [as a workaround for a MySQL issue](https://github.com/wagtail/wagtail/pull/6999), and is immediately followed by changing it to a TextField where max_length doesn't apply. Other than the addition of `default=""`, the field definition here _should_ have matched the previous state, namely [the one set in migration 0005](https://github.com/wagtail/wagtail/blob/main/wagtail/embeds/migrations/0005_specify_thumbnail_url_max_length.py) - but the max_length was missed here.

Marking this as for 2.13.4 - while 2.14 is coming pretty soon, we ought to put out a 2.12.x and 2.13.x release with this fix as it's a potential blocker for people doing point-by-point upgrades.

**Comment by gasman** [comment]
This fix is now released as 2.12.6 and 2.13.4.

**Comment by marts** [comment]
@gasman much appreciated, thanks
